src/logical/program_analysis/analysis.rs

```diff
@@ -480,21 +498,11 @@ impl Program {
         }
 
         for rule in self.rules() {
-            for atom in rule.body() {
-                // check for negation
-                if matches!(atom, Literal::Negative(_)) {
-                    return Err(RuleAnalysisError::UnsupportedFeatureNegation);
-                }
-
-                // check for consistent predicate arities
-                let arity = atom.terms().len();
-                if arity != *arities.entry(atom.predicate()).or_insert(arity) {
-                    return Err(RuleAnalysisError::UnsupportedFeaturePredicateOverloading);
-                }
```

Comment:
Why drop this? What code replaces this check?

Reply:
Moved to the for loop below